lib/ExternalModuleFactoryPlugin.js

```diff
@@ -2,20 +2,21 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var ExternalModule = require("./ExternalModule");
+"use strict";
 
-function ExternalModuleFactoryPlugin(type, externals) {
-	this.type = type;
-	this.externals = externals;
-}
-module.exports = ExternalModuleFactoryPlugin;
+const ExternalModule = require("./ExternalModule");
+
+class ExternalModuleFactoryPlugin {
+	constructor(type, externals) {
+		this.type = type;
+		this.externals = externals;
+	}
 
-ExternalModuleFactoryPlugin.prototype.apply = function(normalModuleFactory) {
-	var globalType = this.type;
-	normalModuleFactory.plugin("factory", function(factory) {
-		return function(data, callback) {
-			var context = data.context;
-			var dependency = data.dependencies[0];
+	apply(normalModuleFactory) {
+		const globalType = this.type;
```

Comment:
I am afraid that it is not possible, since we are using `=>` here the context will change. Either to remove `arrow` or to refer it via `another const/variable`